src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -156,11 +157,8 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         // DON'T CALL _InitializeTerminal here - wait until the swap chain is loaded to do that.
 
         // Subscribe to the connection's disconnected event and call our connection closed handlers.
-        _connectionStateChangedRevoker = _connection.StateChanged(winrt::auto_revoke, [weakThis = get_weak()](auto&& /*s*/, auto&& /*v*/) {
-            if (auto strongThis{ weakThis.get() })
-            {
-                strongThis->_ConnectionStateChangedHandlers(*strongThis, nullptr);
-            }
+        _connectionStateChangedRevoker = _connection.StateChanged(winrt::auto_revoke, [this](auto&& /*s*/, auto&& /*v*/) {
+            _ConnectionStateChangedHandlers(*this, nullptr);
         });
     }

```

Comment:
If we _cannot_ do this because the argument types don't match up, we should 100% just make the member function's arguments match :smile:

Reply:
This one isn't working, because `_ConnectionStateChangedHandlers` wants a `TermControl` object but WinRT is trying to provide something implementing `ITerminalConnection` to it. This is probably because the `{ this, pointer-to-member }` form is only using `this` in conjunction with accessing the member function and not using `this` as a function argument. Instead WinRT is wanting to pass the object owning the event, in this case a `ITerminalConnection`, as the sender argument to the member function.
It's such a clean form, but sadly it only applies to events following a very specific format.

Reply:
Oh, I see -- this isn't a local handler, it is an event _forwarder_. Okay, yeah, no, i retract my whole comment. Thank you. :)

Reply:
("forward the connection's event _as my event_")